xformers/components/multi_head_dispatch.py

```diff
@@ -146,6 +146,18 @@ def forward(
         B, S_Q, _ = query.size()  # Batch x Sequence x Embedding (latent)
         _, S_K, _ = key.size()  # K, Q's sequence length could differ
 
+        # Catch different query and key length but a causal attention
+        if S_Q != S_K:
```

Comment:
Makes it a little more understandable I think, catch early a combination which is not supported